From: Mike Turquette <mturquette@ti.com>
Date: Mon, 23 May 2011 17:58:38 +0000 (-0500)
Subject: OMAP: clock: mute WARN when searching for fieldval
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=515872b71eed708031bdf33d46c44eb5d992a9bd

OMAP: clock: mute WARN when searching for fieldval

This warn is hit every time when entering DPLL cascading.  This is due to a
workaround implemented in DPLL cascading code which must circumvent the
clock framework to change l4_wkup_iclk's parent from SYS_CK to DPLL_ABE's LP
output, and back again.

During low power MP3 playback it is observed that this warn can disrupt
playback by causing a rapid exit/re-entry of the DPLL cascading mode.

Change-Id: I2f65abed7f886a6d7fd5ab5a85b76aedfd77d364
Signed-off-by: Mike Turquette <mturquette@ti.com>
---

Index: kernel/arch/arm/mach-omap2/clkt_clksel.c
===================================================================
--- kernel.orig/arch/arm/mach-omap2/clkt_clksel.c	2011-05-27 23:13:47.777100173 +0900
+++ kernel/arch/arm/mach-omap2/clkt_clksel.c	2011-05-27 23:27:07.168388051 +0900
@@ -188,6 +188,10 @@
 			break;
 	}
 
+	/* 
+	 * OMAP: clock: mute WARN when searching for fieldval .
+	 * Comment: Changes from WARN to pr_debug;
+	 */
 	if (!clkr->div) {
 		/* This indicates a data error */
 		pr_debug("clock: Could not find fieldval %d for clock %s parent "
